src/cpp/src/llm_pipeline.cpp

```diff
@@ -96,6 +96,10 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
     ) override {
         auto start_time = std::chrono::steady_clock::now();
         GenerationConfig config = (generation_config.has_value()) ? *generation_config : m_generation_config;
+        // If eos_token_id was not provided, take value from default m_generation_config
+        if (config.eos_token_id == -1)
+            config.set_eos_token_id(m_generation_config.eos_token_id);
+        config.validate();
```

Comment:
let's update `validate` that `stop_token_ids` contains `eos_token_id` if it's not `-1`